docs/user-guide/meta-models.md

```diff
@@ -177,6 +177,50 @@ This transformer also has use-cases beyond fairness. You could use this transfor
 
+## Hierarchical Prediction
+
+!!! info "New in version 0.8.0"
+
+Very closely related to `GroupedPredictor` is the [`HierarchicalPredictor`][hierarchical-predictor-api] meta estimator and the two task specialized classes [`HierarchicalClassifier`][hierarchical-classifier-api] and [`HierarchicalRegressor`][hierarchical-regressor-api].
+
+These estimators fit a separate base estimator for each group in the input data in a hierarchical manner. This means that an estimator is fitted for each subsequential level in the group columns.
+
+### Difference with `GroupedPredictor`
+
+In practice what does that mean? There are a few main differences between hierarchical and grouped meta estimators:
+
+1. The first difference is the fallback method: hierarchical estimators have a fallback method that can be set to either "parent" or "raise". If set to "parent", the estimator will recursively fall back to the parent group in case the group value is not found during `.predict()`.
+
+    !!! warning
+        As a consequence of this, the **order of groups matters** and potentially a combinatoric number of estimators are fitted, one for each unique combination of group values and each level, including a global one.
+
+2. `HierarchicalClassifier` is meant to properly handle shrinkage for classification tasks - however this requires that the base estimator implements a `.predict_proba()` method.
+3. While `GroupedPredictor` is meant to be used directly, `HierarchicalPredictor` is meant to be used as a base class for other estimators such as `HierarchicalClassifier` and `HierarchicalRegressor`.
+
+## Shrinkage Functions
+
+scikit-lego provides a set of shrinkage functions that can be used to shrink the predictions of a model in [Grouped Prediction] and [Hierarchical Predictions].
+
+The following shrinkage are available out of the box:
+
+- [`"constant"`][constant-shrinkage]: The augmented prediction for each level is the weighted average between its prediction and the augmented prediction for its parent.
+- [`"equal"`][equal-shrinkage]: Each group is weighed equally.
+- [`"min_n_obs"`][min-n-obs-shrinkage]: Use only the smallest group with a certain amount of observations.
+- [`"relative"`][relative-shrinkage]: Weigh each group according to its size.
+
+Additionally to the built-in shrinkage functions, it is possible to provide a custom shrinkage function to the `GroupedPredictor` and `HierarchicalPredictor` classes.
+
+Such function should takes a list of group sizes and returns an array of the same size with the weights (positive values) for each group.
+
+```py title="Custom shrinkage function"
+import numpy as np
+
+def random_shrinkage_function(group_sizes):
+    """A custom random shrinkage function"""
+    a = np.random.uniform(low=0., high=1., size=len(group_sizes))
+    return a / a.sum()
```

Comment:
Maybe something with exponential smoothing instead? A random number as a shrinkage function doesn't feel like it's tangible or real-life enough.